### DISCUSSION TITLE
Draw with Cairo in an image

### DISCUSSION THREAD

**Discussion by rchastain** [mse-org/mseide-msegui#83] [General]
Hello!

I am trying (with no success until now) to draw with Cairo in an image, and draw the image to the screen. The goal would be to have a multiplatform way to draw with Cairo.

The idea is to use **cairo_image_surface_create_for_data** to create the Cairo surface. (I managed to do it with Lazarus and with FreeBASIC, so I know it is possible.)

I post here my (non working) project, in case someone has an idea.

[cairo-experiment.zip](https://github.com/mse-org/mseide-msegui/files/14732930/cairo-experiment.zip)

**Comment by rchastain** [comment]
Here is a Lazarus example, in case someone is interested. I had found the idea [here](https://github.com/sechshelme/Lazarus-SDL_2.0-Examples/tree/main/Cairo/LCL_cairo). (Thanks to @sechshelme.)

[lazarus-cairo_image_surface_create_for_data.zip](https://github.com/mse-org/mseide-msegui/files/14732957/lazarus-cairo_image_surface_create_for_data.zip)

**Comment by fredvs** [comment]
Hello Roland.

I am absolutely beginner in Cairo.
I have to study it.

But what about MSEgui's best friend BGRAbitmap?
Should it not be easier using  a BGRABitmap vs a imagety?
This because msegui can deal in both side with BGRABitmap.

**Comment by fredvs** [comment]
Re-hello Roland.

I dont understand what is the goal, is it to create a memorystream from the cairo surface?

**Comment by rchastain** [comment]
Hello Fred! Thanks for your answer. Indeed, I could use BGRABitmap. So, please, don't waste your time on this. As soon as I find a solution, I will post it here. ;)

In my non working project, the image saved to PNG (a red square) should be drawn on the screen. But the square on the screen is black.
